src/modules/auth/controllers/het-archief.controller.ts

```diff
@@ -127,23 +127,29 @@ export class HetArchiefController {
 
 			SessionHelper.setIdpUserInfo(session, Idp.HETARCHIEF, ldapUser);
 
+			const apps = ldapUser?.attributes?.apps ?? [];
+			const organisationId = isEmpty(ldapUser?.attributes?.o)
+				? null
+				: ldapUser.attributes.o[0];
+			const organisation = await this.organisationService.findOrganisationBySchemaIdentifier(
+				organisationId
+			);
 			let archiefUser = await this.usersService.getUserByIdentityId(
-				ldapUser.attributes.entryUUID[0]
+				ldapUser.attributes.entryUUID[0],
+				organisation
 			);
```

Comment:
This conflates getting something from the database, and setting something that we already know.
I would separate these in separate pieces:
```
this.usersService.getUserByIdentityId(ldapUser.attributes.entryUUID[0]) // Which internally also fetches the organisation if it is available in the avo database

const organisationId = isEmpty(ldapUser?.attributes?.o)
	? null
	: ldapUser.attributes.o[0];
const organisation = await this.organisationService.findOrganisationBySchemaIdentifier(
	organisationId
);
archiefUser.organisationId = organisationId;
archiefUser.organisationName = organisation .name;
```

That way we still have a simple function that fetches the user from the database by id (Single-responsibility principle)
and we augment that user info with the new info we received from the login ldap info afterwards.